pkg/gotohelm/transpiler.go

```diff
@@ -526,50 +526,68 @@ func (t *Transpiler) transpileExpr(n ast.Expr) Node {
 		untyped := [3]string{"_", n.Op.String(), "_"}
 		typed := [3]string{t.typeOf(n.X).String(), n.Op.String(), t.typeOf(n.Y).String()}
 
+		f := func(op string) func(a, b Node) Node {
```

Comment:
never seen this before, how does it work?

Reply:
Function wraps a closure that would be otherwise repeated in the map. That implementation was previously declared https://github.com/redpanda-data/helm-charts/blob/b7a118bedfd652769bfc8c0a5fda58c0aa397917/pkg/gotohelm/transpiler.go#L561-L564